core/src/main/scala/kafka/server/DelayedOperation.scala

```diff
@@ -370,6 +383,20 @@ final class DelayedOperationPurgatory[T <: DelayedOperation](purgatoryName: Stri
 
       purged
     }
+
+    def forceComplete(): Int = {
```

Comment:
The current behavior in DelayedProduce is that if forceComplete() is called, all pending partitions will return a  REQUEST_TIMED_OUT error. However, in the case when the coordinator has changed, we should return a NOT_COORDINATOR error instead.